### PR TITLE
CMR-7853:Add email-subscription-processing job management endpoints

### PR DESCRIPTION
**Pull request by brianalexander**
# Overview
What is the feature?
Add the ability to start/stop the sending of subscription emails if necessary.

What is the Solution?
Add admin endpoints for managing the email subscription job (enable, disable, get state). 

What areas of the application does this impact?
Ingest /jobs routes and the email subscription processing job config (explicitly define job-key)

# Checklist
- [x] I have performed a self-review of my code.
- [x] I have added tests that prove my code is effective or that my feature works.
- [x] I have verified that my change will not break clients.
- [x] I have verified that my code addresses all of the acceptance criteria of the ticket. OR I have explained my motivation for submitting this PR
- [x] I have verified that my code follows CMR Coding Conventions. See the CMR [Coding Style Guidelines](https://wiki.earthdata.nasa.gov/display/CMR/Coding+Style+Guidelines)). 
- [x] I have verified that my code will not negatively impact performance or stability, if applicable.
- [x] I have commented my code, particularly in hard-to-understand areas. 
- [x] I have made corresponding documentation changes if applicable. 
- [x] I have verified that my changes generate no new warnings.

### CODE REVIEW

**Review by jaybarra — APPROVED** [review]

File: ingest-app/README.md [hunk 1]
```diff
@@ -327,6 +330,28 @@ Sends subscription emails covering granules that were updated in the specified r
 ```bash
 curl -i -XPOST -H "Echo-Token: XXXX" %CMR-ENDPOINT%/jobs/trigger-email-subscription-processing?revision-date-range=2011-01-01T10:00:00Z,2012-01-01T10:00:00Z
 ```
+
+### <a name="email-subscription-processing-job-state"></a> Run Subscription Job Over Granule Revision Date Range
```
Comment (ygliuvt) [inline ingest-app/README.md:334]: Update the description of the endpoint.

File: ingest-app/README.md [hunk 2]
```diff
@@ -327,6 +330,28 @@ Sends subscription emails covering granules that were updated in the specified r
...
+
+Returns a JSON response containing the current state of the email-subscription-processing job in the `state` attribute.  The state value corresponds with possible org.Quartz.Trigger.TriggerState values, with NORMAL being enabled and PAUSED being disabled.
+```bash
+curl -i -XGET -H "Echo-Token: XXXX" %CMR-ENDPOINT%/jobs/email-subscription-processing-job-state
+```
+
+### <a name="enable-email-subscription-processing-job"></a> Run Subscription Job Over Granule Revision Date Range
```
Comment (ygliuvt) [inline ingest-app/README.md:341]: The description is wrong. Update it.

File: ingest-app/README.md [hunk 3]
```diff
@@ -327,6 +330,28 @@ Sends subscription emails covering granules that were updated in the specified r
...
+
+Sets the email-subscription-processing job to an enabled state.  If it is already enabled, there is no change.
+```bash
+curl -i -XPOST -H "Echo-Token: XXXX" %CMR-ENDPOINT%/jobs/enable-email-subscription-processing-job
+```
+
+### <a name="disable-email-subscription-processing-job"></a> Run Subscription Job Over Granule Revision Date Range
```
Comment (ygliuvt) [inline ingest-app/README.md:348]: Update the description.

File: ingest-app/src/cmr/ingest/api/routes.clj [hunk 4]
```diff
@@ -82,6 +84,21 @@
            {ctx :request-context params :params}
            (acl/verify-ingest-management-permission ctx :update)
            (subscriptions-helper/trigger-email-subscription-processing ctx params)
+           {:status 200})
+     (GET  "/email-subscription-processing-job-state"
+           {ctx :request-context}
+           (acl/verify-ingest-management-permission ctx :update)
```
Comment (ygliuvt) [inline ingest-app/src/cmr/ingest/api/routes.clj:90]: Shouldn't need extra permission to check the processing state.

Comment (brianalexander) [reply ingest-app/src/cmr/ingest/api/routes.clj]: Removed the permission for this endpoint

**Review by ygliuvt — APPROVED** [review]

**Pull request merged**